### PR TITLE
update submitter guide to use v4.1 schema

### PR DESCRIPTION
**Pull request by NikkyXO**

### CODE REVIEW

**Review by spbnick — CHANGES_REQUESTED** [review]
Hey, very nice start, @NikkyXO!

I have one request inline, and do you think you could try adding sections on "issues" and "incidents" as well? If not, that's OK, the page would already be much better.

Here's an announcement of this schema I made a while back, which could help: https://lore.kernel.org/kernelci/95765418-69ce-54de-93ed-25e29e6c08ea@redhat.com/T/

The schema didn't turn out to be exactly like described there, but it's in the same spirit, and explains a lot. If you'd like, we could also have a quick video call, where I could help you figure it out.

File: doc/submitter_guide.md [hunk 1]
```diff
@@ -153,9 +155,28 @@ checkout with one build and one test:
...
+    ],
+    "incident": [
+        {
+            "id": "submitter:1084645810",
+            "issue_id": "submitter:956769",
+            "origin": "submitter",
+            "issue_version": 0,
+            "build_id": "submitter:956769",
+            "test_id": "submitter:114353810",
+            "present": "true"
```
[the remark this answers is not in the document]
Comment (NikkyXO) [reply doc/submitter_guide.md:174]: alright @spbnick , i will set up a google meet if that's okay, and what time will be convenient?

[the remark this answers is not in the document]
Comment (NikkyXO) [reply doc/submitter_guide.md:174]: Alright, Noted. i will send the meet link on slack by tomorrow

[the remark this answers is not in the document]
Comment (NikkyXO) [reply doc/submitter_guide.md]: > These have fields which are not strictly required, as mentioned in the text just above. Could you remove them? The only required fields are listed in the v4.1 schema under `"required"` for each corresponding object.

@spbnick i have removed non required properties from the objects in the schema

[the remark this answers is not in the document]
Comment (NikkyXO) [reply doc/submitter_guide.md]: > These have fields which are not strictly required, as mentioned in the text just above. Could you remove them? The only required fields are listed in the v4.1 schema under `"required"` for each corresponding object.

@spbnick i have removed non required properties from the objects in the schema

[the remark this answers is not in the document]
Comment (spbnick) [reply doc/submitter_guide.md]: Thank you! The change itself looks good. Could you now squash your commits and rewrite the commit message following the [commit guidelines](https://kernelci.org/docs/kcidb/developer_guide/#commits)?

**Comment by spbnick** [comment]
Looks like you accidentally brought in some unrelated changes, perhaps when rebasing? Could you cleanup the commit, so it only contains the docs change, and remove the unrelated text from the commit message? Reach out on our `#outreachy` Slack channel, if you need help with that, and somebody will help you :heart: 

Also, your documentation change doesn't actually remove anything, as compared to what we have in `main`, so the current subject of "remove non-required properties from objects" doesn't match what you're doing. Something like "docs: Update submitter guide to v4.1 schema" would work better here.

**Pull request closed without merging**